web/src/App.js

```diff
@@ -268,71 +229,111 @@ class App extends Component {
         <Route exact path="/records" render={(props) => this.renderSigninIfNotSignedIn(<RecordListPage account={this.state.account} {...props} />)} />
         <Route exact path="/records/:organizationName/:recordName" render={(props) => this.renderSigninIfNotSignedIn(<RecordEditPage account={this.state.account} {...props} />)} />
         <Route exact path="/access" render={(props) => this.renderSigninIfNotSignedIn(<GuacdPage account={this.state.account} {...props} />)} />
+        <Route exact path="/my-assets" render={(props) => this.renderSigninIfNotSignedIn(<AssetListPage account={this.state.account} {...props} />)} />
+        <Route exact path="/workbench" render={(props) => this.renderSigninIfNotSignedIn(<AssetWorkbench account={this.state.account} {...props} />)} />
       </Switch>
     );
   }
 
   isWithoutCard() {
-    return Setting.isMobile() || window.location.pathname === "/chat";
+    return Setting.isMobile() || this.isHiddenHeaderAndFooter() || window.location.pathname === "/chat";
+  }
+
+  isHiddenHeaderAndFooter(uri) {
+    if (uri === undefined) {
+      uri = this.state.uri;
+    }
+    const hiddenPaths = ["/workbench", "/access"];
+    if (hiddenPaths.includes(uri)) {
+      return true;
+    }
   }
 
   renderContent() {
+    return (
+      <Layout id="parent-area">
+        {this.renderHeader()}
+        <Content style={{display: "flex", flexDirection: "column"}}>
+          {this.isWithoutCard() ?
+            this.renderRouter() :
+            <Card className="content-warp-card">
+              {this.renderRouter()}
+            </Card>
+          }
+        </Content>
+        {this.renderFooter()}
+      </Layout>
+    );
+  }
+
+  renderHeader() {
+    if (this.isHiddenHeaderAndFooter()) {
+      return null;
+    }
+
+    const showMenu = () => {
+      this.setState({
+        menuVisible: true,
+      });
+    };
+
     const onClick = ({key}) => {
-      // eslint-disable-next-line react/prop-types
-      this.props.history.push(key);
+      if (Setting.isMobile()) {
```

Comment:
The diff is not easy to follow here. What does this part of code do?

Reply:
`this.props.history.push(key);` casuse a bug that when you click the link that open in a new tab, the current url will also be update.